File: frontend/src/services/kubernetes.ts
import api from './api';

export interface K8sNamespace {
  name: string;
  status: string;
  age: string;
}

export interface K8sDeployment {
  id?: string;
  name: string;
  namespace: string;
  replicas?: number;
  ready?: string;
  up_to_date?: string;
  available?: string | number;
  image?: string;
  status?: string;
  age?: string;
}

export interface K8sPod {
  id?: string;
  name: string;
  namespace: string;
  deployment?: string;
  ready?: string;
  status: string;
  restarts: string | number;
  node?: string;
  age: string;
}

export interface K8sService {
  name: string;
  namespace: string;
  type: string;
  cluster_ip: string;
  external_ip: string;
  ports: string;
}

export const getNamespaces = async (): Promise<K8sNamespace[]> => {
  const response = await api.get('/kubernetes/namespaces');
  return response.data;
};

export const getDeployments = async (namespace?: string): Promise<K8sDeployment[]> => {
  const response = await api.get('/kubernetes/deployments', {
    params: { namespace },
  });
  return response.data;
};

export const getPods = async (namespace?: string): Promise<K8sPod[]> => {
  const response = await api.get('/kubernetes/pods', {
    params: { namespace },
  });
  return response.data;
};

export const getServices = async (): Promise<K8sService[]> => {
  const response = await api.get('/kubernetes/services');
  return response.data;
};

export const scaleDeployment = async (name: string, replicas: number, namespace: string = 'devops-control-center'): Promise<any> => {
  const response = await api.post(`/kubernetes/deployments/${name}/scale`, {
    replicas,
    namespace,
  });
  return response.data;
};

export const restartPod = async (name: string, namespace: string = 'devops-control-center'): Promise<any> => {
  const response = await api.post(`/kubernetes/pods/${name}/restart`, null, {
    params: { namespace },
  });
  return response.data;
};

export const getPodLogs = async (name: string, namespace: string = 'devops-control-center'): Promise<{ logs: string }> => {
  const response = await api.get<{ logs: string }>(`/kubernetes/pods/${name}/logs`, {
    params: { namespace },
  });
  return response.data;
};

File: backend/app/api/v1/kubernetes.py
from typing import Any, List, Dict
from fastapi import APIRouter, Depends
from app.api import deps
from app.models.user import User

router = APIRouter()

@router.get("/namespaces")
def get_kubernetes_namespaces(
    current_user: User = Depends(deps.get_current_active_user),
) -> List[Dict[str, Any]]:
    return [
        {"name": "default", "status": "Active", "age": "120d"},
        {"name": "kube-system", "status": "Active", "age": "120d"},
        {"name": "production", "status": "Active", "age": "45d"},
        {"name": "staging", "status": "Active", "age": "45d"},
    ]

@router.get("/deployments")
def get_kubernetes_deployments(
    current_user: User = Depends(deps.get_current_active_user),
) -> List[Dict[str, Any]]:
    return [
        {"name": "backend-api", "namespace": "production", "ready": "3/3", "up_to_date": "3", "available": "3", "age": "12d"},
        {"name": "frontend-ui", "namespace": "production", "ready": "2/2", "up_to_date": "2", "available": "2", "age": "12d"},
        {"name": "redis-cache", "namespace": "production", "ready": "1/1", "up_to_date": "1", "available": "1", "age": "40d"},
    ]

@router.get("/pods")
def get_kubernetes_pods(
    current_user: User = Depends(deps.get_current_active_user),
) -> List[Dict[str, Any]]:
    return [
        {"name": "backend-api-7c98d6-xk4p", "namespace": "production", "ready": "1/1", "status": "Running", "restarts": "0", "age": "2d"},
        {"name": "backend-api-7c98d6-yt2m", "namespace": "production", "ready": "1/1", "status": "Running", "restarts": "0", "age": "2d"},
        {"name": "backend-api-7c98d6-zw9l", "namespace": "production", "ready": "1/1", "status": "Running", "restarts": "1", "age": "2d"},
        {"name": "frontend-ui-5f4c8b-abc1", "namespace": "production", "ready": "1/1", "status": "Running", "restarts": "0", "age": "1d"},
    ]

@router.get("/services")
def get_kubernetes_services(
    current_user: User = Depends(deps.get_current_active_user),
) -> List[Dict[str, Any]]:
    return [
        {"name": "backend-svc", "namespace": "production", "type": "ClusterIP", "cluster_ip": "10.96.0.10", "external_ip": "<none>", "ports": "8000/TCP"},
        {"name": "frontend-svc", "namespace": "production", "type": "LoadBalancer", "cluster_ip": "10.96.0.11", "external_ip": "34.120.45.67", "ports": "80:31234/TCP"},
    ]

@router.post("/deployments/{name}/scale")
def scale_kubernetes_deployment(
    name: str,
    replicas: int,
    current_user: User = Depends(deps.get_current_active_user),
) -> Dict[str, Any]:
    return {"message": f"Deployment {name} scaled to {replicas} replicas."}

@router.post("/pods/{name}/restart")
def restart_kubernetes_pod(
    name: str,
    current_user: User = Depends(deps.get_current_active_user),
) -> Dict[str, Any]:
    return {"message": f"Pod {name} deleted. ReplicaSet will recreate it."}
